### ISSUE TITLE
parameter initializer wrongly references variable declared in function body

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#28474]
**TypeScript Version:**  3.2.0-dev.20181101

**Search Terms:**

**Code**

```ts
let baz = 2;
function foo(bar = () => baz) {
    var baz = 1;
    return bar();
}
```

**Expected behavior:**
No error.
`baz` in the initializer references the outer `let baz = 2`.
`let baz = 1` is marked as unused.

**Actual behavior:**
Error `Initializer of parameter 'bar' cannot reference identifier 'baz' declared after it.`

At runtime: calling `foo()` returns `2`, which shows that declarations in the function body are not in scope of parameter initializers.

Note that other parameters **are** in scope and can be used inside other parameters' initializer.

**Playground Link:** https://agentcooper.github.io/typescript-play/#code/DYUwLgBARghgXhAvBATAbgFADMCuA7AYzAEsB7PCLU0gClgCckIaBKJAPmnjYG8MIBEAG4xGsBMgCMmQRHrgc9Cg1aYAvkA

**Related Issues:**
#22769
/cc @Kingwl maybe you want to give this a shot as well (I hope this doesn't cause as much trouble as the last one 😆)

**Comment by ajafff** [comment]
Continuing discussion from https://github.com/Microsoft/TypeScript/pull/28501#issuecomment-438506385

@weswigham @Kingwl 

> I'm pretty sure that error reflects the behavior of our actual down level emit rather than the spec.

That's correct, but we already had that discussion in #22769. I'd argue that the emit should be changed to follow the spec. However, in https://github.com/Microsoft/TypeScript/issues/22769#issuecomment-411138857 there's the decision to change the lookup for ES6 and emit an error for ES5 and below (which means no change for that target).

That means it's still a bug if target is ES6 or above and still needs fixing.

**Comment by Kingwl** [comment]
ahhhhhh, I'll reopen that

**Comment by sandersn** [comment]
This is fixed in 3.8.
